packages/@aws-cdk/core/lib/assets.ts

```diff
@@ -23,6 +23,14 @@ export interface FileAssetSource {
 }
 
 export interface DockerImageAssetSource {
+  /**
+   * The id for the image.
+   *
+   * This will be used to construct the repository name if `repositoryName`
+   * is not specified.
+   */
+  readonly id: string;
```

Comment:
This was intentionally omitted since as part of our work on CI/CD (#3437) we are planning to push all images into a single ECR.

Reply:
You mean you want to maintain this "one repository per stack" behavior for now? This means changing the lifecycle policy

What about layer caching/docker build times in CI with this approach?

Reply:
No, I am okay with trying to fix this, but just not with extending the API surface in order to do that